### PR TITLE
Adjust to use javascript_tag and take html_options

### LINKED ISSUE #8: Proposal: Allow nonces?
hey, was wondering -- we at DCAF Engineering are interested in using this, but have a pretty strict content security policy that prohibits running inline js without whitelisting it with a sha or a nonce. We was mulling over submitting a PR to this that would take a nonce value as an argument, but wanted to check and make sure this was something you all were interested in first.

If you are, our team can whip something up real quick, if that sounds good to you?

### PR DESCRIPTION
**Pull request by colinxfleming**
Hey @nikolalsvk , took a stab at this, interested in your thoughts. After some prodding, the most generally useful way to handle this seemed like adjusting the template to use the `javascript_tag` helper in rails. (I'm guessing that's okay because of your use of `content_for`, but please correct me if I'm wrong!) In a nutshell, this subs out the `<script>` for `javascript_tag html_options`, which lets us pop a hash of html options in, and which fixes my particular problem while also having some utility for anyone else who needs to mash in some html options.

See this commit for an example in action: https://github.com/DCAFEngineering/dcaf_case_management/commit/d0e3631c1281aa7df7552c87c27742d34a1b8a2a We add the html option `nonce: content_security_policy_script_nonce`, and it spits out the following into the view itself from the function `<%= render_async patients_report_path(timeframe: 'weekly'), nonce: content_security_policy_script_nonce %>` --

```
<script nonce="lWmnV6eYicpt+oyOfcShYINsz0b70iR+Q1mohZqNsdg=">
//<![CDATA[

    (function($){
      $.ajax({
          url: "/reports/weekly",
        })
        .done(function(response, status) {
          $("#render_async_18b8a6cd161499117471").html(response);
        })
        .fail(function(response, status) {
          $("#render_async_18b8a6cd161499117471").html(response);
        })
        .always(function(response, status) {
          $("#render_async_18b8a6cd161499117471_spinner").hide();
        });
    }(jQuery));

//]]>
</script>
```

I'm admittedly an rspec infant, but if this looks good to you I can take a whack at adding a test as well, let me know.

Please let me know if I can clarify anything, whether I should bump a changelog somewhere, or any other housekeeping stuff. Thanks much!

Fixes #8 

cc @tingaloo

### CODE REVIEW

**Comment by nikolalsvk** [comment]
Awesome stuff @colinxfleming!

Do you have an idea on how we could test this using RSpec? :)

Also, you can bump the version of this gem in [version file](https://github.com/renderedtext/render_async/blob/master/lib/render_async/version.rb) before merging

P.S. If you could add a short paragraph in README explaining how and when to pass in `html_options` so other users can benefit from it 🙇

**Comment by colinxfleming** [comment]
re: rspec, haha fair. I might take a stab at an everything's okay alarm just to confirm that the function doesn't choke and that the result has, like, a url, a spinner, and whatever html options. (Though I might wind up punting!)

Copy that on the html options and bumping gem version. 💯 I'll tackle next time I have a free minute, should be in the next day or two.

**Comment by nikolalsvk** [comment]
@colinxfleming If you could to that sort of check in RSpec I'd be more than grateful, if not, I can add it too.

Great news for README part and bumping version, can't wait to see it :)

**Comment by colinxfleming** [comment]
Chucked up readme tweaks and version bump, I'll take a stab at the rspec soon-ish though it might be a few days.

File: README.md [hunk 1]
```diff
@@ -75,6 +75,51 @@ And then execute:
...
+## Advanced usage
+
+`render_async` takes two arguments, `path` and `html_options`.
+
+* `path` is the ajax-capable controller action you're looking to call via `get`. e.g. `comments_stats_path`, `posts_path`, etc.
+* `html_options` is an optional hash that gets passed to a rails `javascript_tag`, to drop html tags into the `script` element.
+
+Example utilizing `html_options` with a `nonce`:
+
+    ```ruby
```
Comment (nikolalsvk) [inline README.md]: Can you just loose the indent here, it doesn't look right in preview :(

screen shot 2017-07-06 at 9 58 47 am

File: README.md [hunk 2]
```diff
@@ -75,6 +75,51 @@ And then execute:
...
+    <%= render_async users_path, nonce: 'lWaaV6eYicpt+oyOfcShYINsz0b70iR+Q1mohZqNaag=' %>
+    ```
+
+Rendered code in the view:
+
+    ```html
```
Comment (nikolalsvk) [inline README.md]: Also, rendering of this HTML code block is broken by indentation

File: README.md (hunk 1 above)
Comment (colinxfleming) [reply README.md]: Fixed in d4734cf

File: README.md (hunk 2 above)
Comment (colinxfleming) [reply README.md]: Fixed in d4734cf (I hope)

**Comment by colinxfleming** [comment]
@nikolalsvk I took an awkward stab at some sort of test 'everything's okay alarm' and wound up getting a whole bunch of nowhere. (Turns out it's tough to import actionview modules outside of the context of rails, haha.) If it's cool with you I'd say probably punt on that for a separate PR and merge as is? Otherwise I can continue to plug away at it, but that puts resolution at 'sooner or later', haha. Let me know how you want to proceed, and thanks again for working with me on this, I really appreciate it. 

(Relatedly, happy to push up the failing tests in a separate PR!)

**Comment by nikolalsvk** [comment]
@colinxfleming Great that you've tried at least something with the RSpec, I knew there was going to be something hard about it. You can push it in the separate PR and maybe I'll continue from where you left :)

Anyways, big thanks for the past commits and improvements, you did an awesome job 👏

**Comment by colinxfleming** [comment]
The pleasure's all mine @nikolalsvk ! Thanks for being down to accept it -- super helpful for keeping our stuff rocksolid secure. I'll spin up a WIP brick when I get a chance 💯

**Pull request merged**